supertokens_python/recipe/emailverification/api/implementation.py

```diff
@@ -13,33 +13,34 @@
 # under the License.
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Dict
+from typing import TYPE_CHECKING, Any, Dict, Union
 
 from supertokens_python.recipe.emailverification.interfaces import (
-    APIInterface, EmailVerifyPostInvalidTokenErrorResponse,
-    EmailVerifyPostOkResponse,
+    APIInterface, CreateEmailVerificationTokenEmailAlreadyVerifiedErrorResult,
+    EmailVerifyPostInvalidTokenErrorResponse, EmailVerifyPostOkResponse,
     GenerateEmailVerifyTokenPostEmailAlreadyVerifiedErrorResponse,
-    GenerateEmailVerifyTokenPostOkResponse, IsEmailVerifiedGetOkResponse)
+    GenerateEmailVerifyTokenPostOkResponse, IsEmailVerifiedGetOkResponse,
+    VerifyEmailUsingTokenOkResult)
 
 if TYPE_CHECKING:
     from supertokens_python.recipe.emailverification.interfaces import (
-        APIOptions, GenerateEmailVerifyTokenPostResponse, IsEmailVerifiedGetResponse, EmailVerifyPostResponse
+        APIOptions
     )
 
 from supertokens_python.recipe.emailverification.types import User
 from supertokens_python.recipe.session.asyncio import get_session
 
 
 class APIImplementation(APIInterface):
-    async def email_verify_post(self, token: str, api_options: APIOptions, user_context: Dict[str, Any]) -> EmailVerifyPostResponse:
+    async def email_verify_post(self, token: str, api_options: APIOptions, user_context: Dict[str, Any]) -> Union[EmailVerifyPostOkResponse, EmailVerifyPostInvalidTokenErrorResponse]:
         response = await api_options.recipe_implementation.verify_email_using_token(token, user_context)
-        if response.is_ok:
+        if isinstance(response, VerifyEmailUsingTokenOkResult):
             if response.user is None:
                 raise Exception("Should never come here")
```

Comment:
This type of checking should not be needed anymore - since `VerifyEmailUsingTokenOkResult` will always contain the user object in it.